packages/router/src/lib/helpers/prepare.ts

```diff
@@ -28,19 +27,24 @@ export const validExpiryBuffer = (buffer: number) => buffer > ONE_DAY_IN_SECONDS
 export const validBidExpiry = (bidExpiry: number, currentTime: number) => bidExpiry > currentTime;
 
 /**
- * Returns the amount * SWAP_RATE to deduct fees when going from sending -> recieving chain to incentivize routing.
+ * Returns the amount * swapRate to deduct fees when going from sending -> recieving chain to incentivize routing.
  *
  * @param amount The amount of the transaction on the sending chain
- * @returns The amount, less fees as determined by the SWAP_RATE
+ * @returns The amount, less fees as determined by the swapRate
  *
  * @remarks
- * Router fulfills on sending chain, so gets `amount`, and user fulfills on receiving chain so gets `amount * SWAP_RATE`
+ * Router fulfills on sending chain, so gets `amount`, and user fulfills on receiving chain so gets `amount * swapRate`
  */
-export const getReceiverAmount = (amount: string, inputDecimals: number, outputDecimals: number): string => {
+export const getReceiverAmount = (
+  amount: string,
+  inputDecimals: number,
+  swapRate: string,
+  outputDecimals: number,
+): string => {
   if (amount.includes(".")) {
     throw new AmountInvalid(amount);
   }
-  return calculateExchangeWad(BigNumber.from(amount), inputDecimals, SWAP_RATE, outputDecimals).toString();
+  return calculateExchangeWad(BigNumber.from(amount), inputDecimals, swapRate, outputDecimals).toString();
```

Comment:
First calculate with swap rate from AMM (hardcoded to 1). Then subtract flat 0.05% fee.